platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ConfigurationRoleUsageIntegrationTest.groovy

```diff
@@ -420,57 +429,143 @@ class ConfigurationRoleUsageIntegrationTest extends AbstractIntegrationSpec impl
+        executer.expectDocumentedDeprecationWarning("Gradle will mutate the usage of configuration customCompileOnly to match the expected usage. This may cause unexpected behavior. Creating configurations with reserved names has been deprecated. This will fail with an error in Gradle 9.0. Create source set custom prior to creating or accessing the configurations associated with it. For more information, please refer to https://docs.gradle.org/current/userguide/building_java_projects.html#sec:implicit_sourceset_configurations in the Gradle documentation.")
         fails 'help'
 
+        then:
+        failure.assertHasDescription("An exception occurred applying plugin request [id: 'java']")
+        failure.assertHasCause("Failed to apply plugin class 'org.gradle.api.plugins.JavaBasePlugin'.")
+        failure.assertHasCause("Gradle cannot mutate the usage of configuration 'customCompileOnly' because it is locked.")
+    }
+
+    def "changing usage #property = #change on detached configurations fails"() {
+        given:
+        buildFile << """
+            def detached = project.configurations.detachedConfiguration()
+
+            assert !detached.canBeConsumed
+            assert detached.canBeResolved
+            assert detached.canBeDeclared
+
+            detached.$property($change)
+        """
+
+        when:
+        fails "help"
+
         then:
         failure.assertHasDescription("A problem occurred evaluating root project '${buildFile.parentFile.name}'.")
         failure.assertHasCause("""Method call not allowed
-  Calling setCanBeResolved(true) on configuration ':$configuration' is not allowed.  This configuration's role was set upon creation and its usage should not be changed.""")
+  Calling $property($change) on configuration ':detachedConfiguration1' is not allowed.  This configuration's role was set upon creation and its usage should not be changed.""")
 
-        where: "a non-exhaustive list of configurations is tested"
-        configuration << ['api', 'implementation', 'compileOnly', 'runtimeOnly', 'archives']
+        where:
+        property | change
+        "setCanBeConsumed" | true
+        "setCanBeResolved" | false
+        "setCanBeDeclared" | false
     }
 
-    def "setting consumable = false fails for consumable configurations added by java plugin"() {
-        given: "a buildscript which attempts to change a configuration's usage"
+    def "changing usage on detached configurations fails when flag is set"() {
+        given:
         buildFile << """
-            plugins {
-                id 'java-library'
-            }
+            def detached = project.configurations.detachedConfiguration()
+
+            detached.canBeResolved = false
+            detached.canBeConsumed = false
+            detached.canBeDeclared = false
+        """
 
+        when:
+        fails('help', "-Dorg.gradle.internal.deprecation.preliminary.Configuration.redundantUsageChangeWarning.enabled=true")
```

Comment:
I would expect changing the usages to fail for each usage even if the flag is not set. 

Redundant usages should emit a warning if the flag is set.